### PR TITLE
vimPlugins.vim-clang-format: init at 2019-05-15

### PR DESCRIPTION
**Pull request by siriobalmelli**
Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

###### Motivation for this change

I daily-drive nixpkgs.neovim on Darwin and Linux, and wanted vim-clang-format plugin

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -576,6 +579,10 @@ self: super: {
     passthru.python3Dependencies = ps: with ps; [ beancount ];
   });
 
+  vim-clang-format = super.vim-clang-format.overrideAttrs(old: {
+    propagatedBuildInputs = [ clang-tools ];
+  });
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix:584]: I would prefer patching, and avoid adding stuff to a user's path

**Review by jonringer — COMMENTED** [review]
also, please follow https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs

There should be two commits

**Comment by siriobalmelli** [comment]
> also, please follow https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs
> 
> There should be two commits

Ok, fixed, thank you.

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (siriobalmelli) [reply pkgs/misc/vim-plugins/overrides.nix]: ok, tracking.

My experience falls a bit short here, and I didn't see any other examples in `overrides.nix` of how one would patch in a dependency like that 😅 

For simplicity's sake I've opted for *just* the standard plugin for now, and I'm betting a large majority of users for this plugin will already have `clang-format` in their environment.

If you're not happy with this approach I'm happy to figure out patching, please point to any docs/examples and I'll got work through it 😄

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 95366 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>60 packages built:</summary>  - vimPlugins.NeoSolarized  - vimPlugins.YouCompleteMe  - vimPlugins.ale  - vimPlugins.coc-fzf  - vimPlugins.coc-go  - vimPlugins.coc-nvim  - vimPlugins.completion-nvim  - vimPlugins.conjure  - vimPlugins.dart-vim-plugin  - vimPlugins.defx-git  - vimPlugins.defx-icons  - vimPlugins.defx-nvim  - vimPlugins.deoplete-fish  - vimPlugins.deoplete-jedi  - vimPlugins.deoplete-khard  - vimPlugins.deoplete-nvim  - vimPlugins.fzf-vim  - vimPlugins.ghcid  - vimPlugins.kotlin-vim  - vimPlugins.lh-vim-lib  - vimPlugins.lightline-vim  - vimPlugins.ncm2-jedi  - vimPlugins.neocomplete-vim  - vimPlugins.neoterm  - vimPlugins.nerdtree  - vimPlugins.neuron-vim  - vimPlugins.nvim-lsp  - vimPlugins.nvim-treesitter  - vimPlugins.onedark-vim  - vimPlugins.palenight-vim  - vimPlugins.syntastic  - vimPlugins.unicode-vim  - vimPlugins.vim-airline  - vimPlugins.vim-bazel  - vimPlugins.vim-beancount  - vimPlugins.vim-clang-format  - vimPlugins.vim-codefmt  - vimPlugins.vim-css-color  - vimPlugins.vim-devicons  - vimPlugins.vim-eunuch  - vimPlugins.vim-fireplace  - vimPlugins.vim-floaterm  - vimPlugins.vim-gitgutter  - vimPlugins.vim-go  - vimPlugins.vim-maktaba  - vimPlugins.vim-monokai  - vimPlugins.vim-one  - vimPlugins.vim-pandoc-syntax  - vimPlugins.vim-plug  - vimPlugins.vim-polyglot  - vimPlugins.vim-signify  - vimPlugins.vim-slime  - vimPlugins.vim-snippets  - vimPlugins.vim-terraform  - vimPlugins.vim-test  - vimPlugins.vim-visual-multi  - vimPlugins.vim-vsnip  - vimPlugins.vim-which-key  - vimPlugins.vimtex  - vimPlugins.vista-vim</details>

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/misc/vim-plugins/overrides.nix]: Hmm, I can't find any great examples either, but I'm fine with refining it in a different PR.

I think most people will figure out that they need `clang-format` available for it to work.

maybe propagating is the solution, and I've been reviewing too many python packages

**Pull request merged**